mlflow/transformers.py

```diff
@@ -854,13 +859,44 @@ def _load_model(path: str, flavor_config, return_type: str, device=None, **kwarg
         component_type = flavor_config[component_type_key]
         conf[component_key] = _load_component(local_path, component_key, component_type)
 
+    if _TORCH_DTYPE_KEY in flavor_config:
+        conf[_TORCH_DTYPE_KEY] = _deserialize_torch_dtype_if_exists(flavor_config)
+
+    for key in _METADATA_PIPELINE_SCALAR_CONFIG_KEYS:
+        if key in flavor_config:
+            conf[key] = flavor_config[key]
+
     if return_type == "pipeline":
         conf.update(**kwargs)
         return transformers.pipeline(**conf)
     elif return_type == "components":
         return conf
 
 
+def _deserialize_torch_dtype_if_exists(flavor_config):
+    """
+    Convert the string-encoded `torch_dtype` pipeline argument back to the correct `torch.dtype`
+    instance value for applying to a loaded pipeline instance.
+    """
+
+    try:
+        import torch
+
+        dtype_mapping = {
+            str(dtype): dtype
+            for name, dtype in torch.__dict__.items()
+            if isinstance(dtype, torch.dtype)
+        }
```

Comment:
we'd better only initialize `dtype_mapping` only once to improve performance. We can make it to be a lazily initialized global variable.

Reply:
Does this work ?
`getattr(torch, dtype_string)`

Reply:
That's what I tried at first, but unfortunately, these aren't module attributes (trying to access them via getattr() from `torch.__init__()` will throw an AttributeError). 

Great idea on memoization. Externalized that type mapping to a function backed by an lru_cache entry and confirmed that we save > 1s per call. Thanks @WeichenXu123 :)